src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

```diff
@@ -158,9 +168,47 @@ public String getDetailsPage(@PathVariable Long submissionId, Model model, final
 		model.addAttribute("analysisType", analysisType);
 		model.addAttribute("mailConfigured", emailController.isMailConfigured());
 
+		if(submission.getAnalysisState()==AnalysisState.ERROR) {
+			model.addAttribute("previousState", getPreviousStateBeforeError(submissionId));
+		}
+
 		return "analysis";
 	}
 
+	/**
+	 * Gets the state of analysis prior to error
+	 *
+	 * @param submissionId {@link Long} identifier for an {@link AnalysisSubmission}
+	 * @return {@link String} State of analysis prior to error
+	 */
+	private AnalysisState getPreviousStateBeforeError(Long submissionId) {
+		AuditReader audit =AuditReaderFactory.get(entityManagerFactory.createEntityManager());
+		AnalysisSubmission previousRevision = null;
+
+		if(audit != null) {
+			// Get revisions from the analysis submission audit table for the submission
+			List auditResultSet = audit.createQuery()
```

Comment:
Is it possible to parameterize this list?

Reply:
Added in 47d5d9d0375e49f3d1f50ccc63a12242db6c13e2

Reply:
Nice, thanks